### PR TITLE
kb(common): Add KB for InPlaceEditor component

### PR DESCRIPTION
**Pull request by dimodi**

### CODE REVIEW

File: knowledge-base/in-place-editor.md [hunk 1]
```diff
@@ -0,0 +1,599 @@
+---
+title: In-Place Editor
```
Comment (dimodi) [inline knowledge-base/in-place-editor.md:2]:
```suggestion
title: In-Place Editor Component
```

File: knowledge-base/in-place-editor.md [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+    * [TimePicker]({%slug components/timepicker/overview%}}) for `TimeOnly`
+* If the `Width` parameter is not set, the In-Place Editor approximately matches the width of its editor components to the current `Value` length. The component uses a `monospace` `font-family` to make this easier.
+* The component features a `ReadOnly` mode that controls the editability, for example, depending on user permissions.
+* The `DisplayFormat` parameter affects the `Value` consistently in both read mode and edit mode.
+* The `Placeholder` parameter provides a helper label that will show when the `Value` is `null` or empty.
+* The `ShowIcon` parameter controls the visibility of an optional [SVG Icon]({%slug common-features-icons%}}#svgicon-component) that hints users about the ability to edit the component `Value`. The parameter is of type `InPlaceEditorShowIcon`, which is a custom enum and must be imported in both `TelerikInPlaceEditor.razor` and all `.razor` files that use `TelerikInPlaceEditor`.
+* To [see invalid state styling and validation messages in Forms]({%slug inputs-kb-validate-child-component%}), pass the respective `ValueExpression` values to the `InPlaceEditor` component.
```
Comment (dimodi) [inline knowledge-base/in-place-editor.md:50]:
```suggestion
* To [see invalid state styling and validation messages in Forms]({%slug inputs-kb-validate-child-component%}), pass the respective `ValueExpression` values to the `InPlaceEditor` component.
* `TelerikInPlaceEditor.razor.css` is a <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/components/css-isolation" target="_blank">CSS isolation file</a>. It depends on a `YourAppName.styles.css` file in `App.razor` to load.
```

File: knowledge-base/in-place-editor.md [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+* `TelerikInPlaceEditor.razor.css` is a <a href="https://learn.microsoft.com/en-us/aspnet/core/blazor/components/css-isolation" target="_blank">CSS isolation file</a>. It depends on a `YourAppName.styles.css` file in `App.razor` to load.
+
+### Example
+
+The features and business logic below can be subject to additional customizations and enhancements.
```
Comment (dimodi) [inline knowledge-base/in-place-editor.md:55]:
```suggestion
The features and business logic below can be subject to additional customizations and enhancements.

Replace `YourAppName` with the actual root namespace of your app.
```

File: knowledge-base/in-place-editor.md [hunk 4]
```diff
@@ -0,0 +1,743 @@
...
+* How to toggle between text content and an editor to allow users to edit something in place?
+
+## Solution
+
+The sample below uses an algorithm which toggles between read-only UI and an editable component on user click and blur.
+
+### How It Works
+
+* `TelerikInPlaceEditor` is a generic component. It supports strings and most value types, including nullable types.
+* Initially, the component renders a clickable [Button]({%slug components/button/overview%}}) with [`Clear` `FillMode`]({%slug button-appearance%}) that shows the current `Value`.
```
Comment (ntacheva) [inline knowledge-base/in-place-editor.md:38]:
```suggestion
* Initially, the component renders a clickable [Button]({%slug components/button/overview%}) with [`Clear` `FillMode`]({%slug button-appearance%}) that shows the current `Value`.
```

**Comment by ntacheva** [comment]
It looks to me that the Show Icon = Hover does not work. As I understand, this configuration is expected to affect the edit icon here:
image

The `Never` option hides the icon but `Hover` does not change anything.

**Comment by ntacheva** [comment]
A couple more points:

- I am concerned about the naming. **Telerik**InPlaceEditor may indicate that it is an actual supported component considering teh naming for ur components and this is not the case.
- Parts of the description are not logically related to the included InPlaceEditor. For example:

image
I find it a bit odd as I don't see the relation between the icons and the time value.

**Comment by dimodi** [comment]
> It looks to me that the Show Icon = Hover does not work. As I understand, this configuration is expected to affect the edit icon here: image
> 
> The `Never` option hides the icon but `Hover` does not change anything.

CSS isolation stylesheet cache. I added a note about this in the KB.

**Review by ntacheva — APPROVED** [review]

**Pull request merged**